### PR TITLE
Bullet list for enum values

### PR DESCRIPTION
**Pull request by emackey**
I think they're easier to read this way.  The example in the README looks more verbose, but enums that have long descriptions per-value are much better.

/cc @HowardWolosky

### CODE REVIEW

File: lib/generateMarkdown.js [hunk 1]
```diff
@@ -405,15 +404,17 @@ function getEnumString(schema, type) {
...
  */
-function getAnyOfEnumString(schema, type) {
+function getAnyOfEnumString(schema, type, depth) {
     var propertyAnyOf = schema['anyOf'];
     if (!defined(propertyAnyOf)) {
         return undefined;
     }
 
     var allowedValues = '';
+    var prefix = '\n' + ('      '.substring(0, depth * 3)) + '* ';
```
Comment (HowardWolosky) [inline lib/generateMarkdown.js]: Might just be me, but I think this:
`('      '.substring(0, depth * 3))`
would probably be a little easier to read as:
`' '.repeat(depth * 3)`

File: lib/generateMarkdown.js [hunk 2]
```diff
@@ -373,29 +373,28 @@ function getPropertySummary(property, knownTypes, autoLink) {
...
     if (!defined(propertyEnum)) {
         // It's possible that the enum value is defined using the anyOf construct instead.
-        return getAnyOfEnumString(schema, type);
+        return getAnyOfEnumString(schema, type, depth);
     }
 
     var propertyEnumNames = schema['gltf_enumNames'];
 
     var allowedValues = '';
+    var prefix = '\n' + ('      '.substring(0, depth * 3)) + '* ';
```
Comment (HowardWolosky) [inline lib/generateMarkdown.js]: Should this be a one-liner that comes out of style?  It seems like `style` should be the unified place where all markup logic is...so, something like `style.getBulletPoint(element, depth)`, and that would both figure out the `prefix` and do the concatenation that you're doing later in `397`.

**Comment by emackey** [comment]
Thanks!  Turns out there was already a `style.bulletItem` that I should have been using.

File: lib/generateMarkdown.js [hunk 3]
```diff
@@ -433,7 +431,7 @@ function getAnyOfEnumString(schema, type, depth) {
             enumString  += " " + enumDescription;
         }
 
-        allowedValues += prefix + enumString;
+        allowedValues += style.bulletItem(enumString, depth).toString();
```
Comment (HowardWolosky) [inline lib/generateMarkdown.js]: Why is the `toString()` necessary here?  Did you mean to put that on the `enumString` instead?

File: lib/generateMarkdown.js [hunk 4]
```diff
@@ -386,15 +386,14 @@ function getEnumString(schema, type, depth) {
...
-    var prefix = '\n' + ('      '.substring(0, depth * 3)) + '* ';
     var length = propertyEnum.length;
     for (var i = 0; i < length; ++i) {
         var element = style.enumElement(propertyEnum[i], type);
         if (defined(propertyEnumNames)) {
             element += " " + propertyEnumNames[i];
         }
 
-        allowedValues += prefix + element;
+        allowedValues += style.bulletItem(element.toString(), depth);
```
Comment (HowardWolosky) [inline lib/generateMarkdown.js]: I took a look at the code for `style.bulletItem` and it looks like it has a bug.  It does all that indentation logic, and then it just drops it at the end (it doesn't do a `+=` when adding the bullet and item content, but rather just an `=`).

So...you should probably fix `style.bulletItem` with this PR, and you may as well simplify the logic to use `.repeat` as opposed to the `for loop`.

File: lib/generateMarkdown.js (hunk 3 above)
Comment (emackey) [inline lib/generateMarkdown.js]: The `toString()`s were leftover from debugging something.  I pulled them out.

**Comment by emackey** [comment]
Thanks for catching these!

**Comment by pjcozzi** [comment]
@HowardWolosky feel free to merge PRs in this repo after you review.

**Pull request merged**